src/utils.ts

```diff
@@ -740,3 +740,22 @@ export const isShallowEqual = <T extends Record<string, any>>(
   }
   return aKeys.every((key) => objA[key] === objB[key]);
 };
+
+// taken from Radix UI
+// https://github.com/radix-ui/primitives/blob/main/packages/core/primitive/src/primitive.tsx
+export const composeEventHandlers = <E>(
+  originalEventHandler?: (event: E) => void,
+  ourEventHandler?: (event: E) => void,
+  { checkForDefaultPrevented = true } = {},
+) => {
+  return function handleEvent(event: E) {
+    originalEventHandler?.(event);
+
+    if (
+      checkForDefaultPrevented === false ||
```

Comment:
explicit check required ?

Reply:
no, I just copied the function verbatim. But why not be explicit.